### PR TITLE
Clarified gas limits 

### PR DESCRIPTION
**Pull request by MadelineMurray**
Clarified gas limits (block gas limit vs transaction gas limit)

### CODE REVIEW

File: docs/Concepts/Transactions/Transaction-Validation.md [hunk 1]
```diff
@@ -12,11 +12,11 @@ validations are repeated when the block including the transaction is imported ex
 exactly right when importing the block. 
 
 When the transaction is added to a block an additional validation is performed to check the transaction gas limit
-is less than the remaining gas limit for the block. After creating a block, the node imports the block and
+is less than the remaining block gas limit. After creating a block, the node imports the block and
 the transaction pool validations are repeated. 
 
 !!! important 
-    The transaction is only added if the entire transaction gas limit is less than the remaining gas 
+    The transaction is only added if the entire transaction gas limit is less than the remaining block gas
```
Comment (NicolasMassart) [inline docs/Concepts/Transactions/Transaction-Validation.md]: seems redundant to repeat "block" here.

File: docs/Reference/API-Methods.md [hunk 2]
```diff
@@ -1673,7 +1673,7 @@ The `eth_estimateGas` call does not send a transaction. You must make a subseque
 **Parameters**
 
 The transaction call object parameters are the same as those for [eth_call](#eth_call), except that in `eth_estimateGas`,
-all fields are optional. Setting a gas limit is irrelevant to the estimation process (unlike transactions, in which gas
+all fields are optional. Setting a transaction gas limit is irrelevant to the estimation process (unlike transactions, in which gas
```
Comment (NicolasMassart) [inline docs/Reference/API-Methods.md]: did you mean "call gas limit"?

File: docs/Concepts/Transactions/Transaction-Validation.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Concepts/Transactions/Transaction-Validation.md]: Removed

File: docs/Reference/API-Methods.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Removed - I think introducing another term is probably more confusing than not specifying transaction here

**Pull request merged**